### PR TITLE
py3-ml-metadata.yaml: Add openssf-options

### PR DESCRIPTION
**Pull request by justinvreeland**

### CODE REVIEW

**Comment by justinvreeland** [comment]
This is a bit annoying. We swapped to overrides with `GCC_SPEC_FILE` because some build systems ignore environment variables setting via common compiler flags.  It seems here we have a case of py3-ml-metadata under certain circumstances ignoring all unknown environment variables.

I can't override the FORTIFY_SROUCE through flags, when I update our OPENSSF_COMPILER_OPTIONS to point to one without fortify source I get a new error: 

```
2024/12/09 16:28:23 WARN   gcc: error: unrecognized command-line option '-Wthread-safety' subpackage=py3.10-ml-metadata uses=py/pip-build-install name="pip build"
2024/12/09 16:28:23 WARN   gcc: error: unrecognized command-line option '-Wself-assign' subpackage=py3.10-ml-metadata uses=py/pip-build-install name="pip build"
2024/12/09 16:28:23 WARN   gcc: error: unrecognized command-line option '-fcolor-diagnostics' subpackage=py3.10-ml-metadata uses=py/pip-build-install name="pip build"
```

While compiling one of the vendors deps. Which is unpleasant because it seems our compiler options are causing Bazel problems by being different depending on if the environment variable is seen by a particular build. And also because protobuf is on the list of projects that we can't build with our options right now: https://github.com/wolfi-dev/os/issues/34075

The only way I've found to get this package to build so far is to create an empty `openssf-compiler-options-null` package that provides `openssf-compiler-options` which is really what we're trying to avoid.

Using the various Bazel options to try and pass additional environment variables does not seem to have an impact and while there's and example of updated dependencies in our patches I haven't found a combination that works for a full build.

**Comment by xnox** [comment]
Please add explicit build-depends on openssf-compiler-options; as i see in the logs that -r5 is getting installed, instead of -r6.

**Comment by xnox** [comment]
Also, in the future, to support https://github.com/chainguard-dev/internal/discussions/3879 i am hoping to do opportunistic inclusion of melange.spec file.

I wonder if we should really create a null.spec file; and make the config file specify the spec file to use (or symlink the active spec file?!) such that one can self-set that; or we can package those as things we can depend on.

Cause in the future, i hope that non-hardened builds, pick up ELF notes automatically still.

**Comment by justinvreeland** [comment]
That's a bit annoying it looks like it's still getting the current version of openssf-compiler-options.

**Comment by justinvreeland** [comment]
Yay!

File: openssf-compiler-options.yaml [hunk 1]
```diff
@@ -57,6 +57,14 @@ test:
         touch foo.c
         gcc -v foo.c &>stderr.log || true
         grep gcc_s stderr.log
+    - name: Ensure etc specfile is used
+      runs: |
+        touch /etc/gcc_spec_file
+        gcc -v 2>&1 | grep "/etc/gcc_spec_file"
+    - name: Ensure GCC_SPEC_FILE is used over "/etc/gcc_spec_file"
+      runs: |
```
Comment (smoser) [inline openssf-compiler-options.yaml:65]: can you create /etc/gcc_spec_file here ? to be clear that it exists ?

Comment (smoser) [reply openssf-compiler-options.yaml:65]: and maybe just rm it above to make the tests more standalone.

File: py3-ml-metadata.yaml [hunk 2]
```diff
@@ -49,6 +50,9 @@ pipeline:
     with:
       series: series
 
+  - runs: |
+      touch /etc/gcc_spec_file
```
Comment (smoser) [inline py3-ml-metadata.yaml:54]: i think i would prefer Setting the environment var to /dev/null.
I'm interested in your thoughts on that versus this.

File: openssf-compiler-options.yaml (hunk 1 above)
Comment (justinvreeland) [reply openssf-compiler-options.yaml:65]: Can do.

File: py3-ml-metadata.yaml (hunk 2 above)
Comment (justinvreeland) [reply py3-ml-metadata.yaml:54]: Doesn't work which is why we have I'm making the file. The build eventually clears all the environment variables.

Comment (justinvreeland) [reply py3-ml-metadata.yaml:54]: Which is to say I also would've preferred the environment variable.

**Review by xnox — APPROVED** [review]

**Pull request merged**